client/layout/index.jsx

```diff
@@ -116,7 +116,9 @@ class Layout extends Component {
 				<BodySectionCssClass group={ this.props.sectionGroup } section={ this.props.sectionName } />
 				<DocumentHead />
 				<QuerySites primaryAndRecent />
-				<QuerySites allSites />
+				{ this.props.currentRoute && this.props.currentRoute !== '/jetpack/connect/authorize' && (
```

Comment:
Might be worth a comment in here as to why we need this crazy work-around.

Reply:
Good call, some explanation added in https://github.com/Automattic/wp-calypso/pull/31277/commits/4550bdb1a1dbc895307f01515ee7ef71e5ac36fb

Reply:
Instead of adding more complex conditionals to an already heavy JSX markup, can we turn the code into a (locally defined) selector and call it in `mapStateToProps`?

```jsx
function routeShouldQueryAllSites( state ) {
  return getCurrentRoute( state ) !== '/jetpack/connect/authorize';
}

connect( state => ( {
  shouldQueryAllSites: routeShouldQueryAllSites( state );
} ) );

// render
{ shouldQueryAllSites && <QueryAllSites /> }
```

The code that decides whether to show the inline help would benefit from a similar update, too.

> this is a change at a framework level - we don't start querying sites before the current route is set.

Yes, avoiding the query when the route is not set yet is needed, because otherwise the [initial `renderLayout`](https://github.com/Automattic/wp-calypso/blob/0ba64da48f35039d665d8366d5ca6d884a520af3/client/boot/project/wordpress-com.js#L80) would trigger it.

The only other way how `<Layout />` is ever rendered is to handle a route. And the current route is set very early by the [`setRouteMiddleware` handler](https://github.com/Automattic/wp-calypso/blob/0ba64da48f35039d665d8366d5ca6d884a520af3/client/boot/common.js#L116-L122).

Reply:
Sounds good - I've moved the condition and the comment to `mapStateToProps` in https://github.com/Automattic/wp-calypso/pull/31277/commits/9f2c2a50d0653dff87988bf691e8136b7451d312 - would appreciate another 👀 , thanks!

Reply:
Yes, that's what I had in mind :+1: